aquatic-invasions.csl

```diff
@@ -161,50 +160,101 @@
+            <else-if type="book" match="any">
+              <choose>
+                <if match="any" variable="edition">
+                  <group delimiter=" ">
+                    <group delimiter=", " suffix=". ">
+                      <text macro="title"/>
                       <text macro="edition"/>
                     </group>
-                  </if>
-                  <else>
-                    <text variable="edition" suffix="."/>
-                  </else>
-                </choose>
-              </group>
-              <group>
-                <choose>
-                  <if match="any" variable="collection-title">
-                    <text variable="collection-title"/>
-                    <group suffix=".">
-                      <label plural="never" text-case="capitalize-first" suffix=" " variable="volume"/>
-                      <text variable="volume"/>
+                    <group delimiter=", ">
+                      <text variable="collection-title"/>
                     </group>
-                  </if>
-                </choose>
-              </group>
-              <group delimiter=", ">
-                <text prefix=" " macro="publisher"/>
-                <text macro="page"/>
-              </group>
-            </if>
+                    <group delimiter=", ">
+                      <text macro="publisher"/>
+                      <text macro="page"/>
+                    </group>
+                  </group>
+                </if>
+                <else>
+                  <group delimiter=", " suffix=". ">
+                    <group delimiter=". ">
+                      <text macro="title"/>
+                      <text variable="collection-title"/>
+                    </group>
+                    <group delimiter=", ">
+                      <group>
+                        <label plural="never" text-case="capitalize-first" suffix=" " variable="volume"/>
+                        <text variable="volume"/>
+                      </group>
+                    </group>
+                  </group>
+                  <group delimiter=", ">
+                    <text macro="publisher"/>
+                    <text macro="page"/>
+                  </group>
+                </else>
+              </choose>
+            </else-if>
             <else-if type="thesis">
               <group delimiter=". ">
                 <text macro="title"/>
-                <group delimiter=", ">
+                <group delimiter=", " suffix=", ">
                   <text variable="genre" strip-periods="true" suffix=" Thesis"/>
                   <text macro="publisher"/>
                 </group>
               </group>
               <text macro="page"/>
             </else-if>
-            <else-if type="chapter paper-conference" match="any">
+            <else-if type="chapter" match="any">
+              <group delimiter=", " suffix=". ">
+                <group delimiter=" ">
+                  <group suffix=". ">
```

Comment:
The issue here is that when I remove the group, it defaults to the delimiter of the preceding group and I need (for this ref type), a period instead of a comma between the title and container-title but a comma between other things. I played around with it for a while and this seemed to be the best way to get that, but again, still new to this!